rust/xaynet-mobile/tests/ffi_test.c

```diff
@@ -74,6 +74,34 @@ static char *test_settings() {
   return 0;
 }
 
+static char *test_global_model() {
+  Settings *settings = xaynet_ffi_settings_new();
+  with_keys(settings);
+  with_url(settings);
+  xaynet_ffi_settings_set_url(settings, "http://localhost:8081");
+
+  Participant *participant = xaynet_ffi_participant_new(settings);
+  mu_assert("failed to create participant", participant != NULL);
+  ModelConfig *model_config = xaynet_ffi_participant_model_config(participant);
+  float* buffer = (float *)malloc(sizeof(float) * model_config->len);
+
+  int err = xaynet_ffi_participant_global_model(NULL, buffer, model_config->data_type, model_config->len);
+  mu_assert("expected participant is null error", err == ERR_NULLPTR);
+
+  err = xaynet_ffi_participant_global_model(participant, NULL, model_config->data_type, model_config->len);
+  mu_assert("expected buffer is null error", err == ERR_NULLPTR);
+
+  err = xaynet_ffi_participant_global_model(participant, buffer, model_config->data_type, model_config->len);
+  mu_assert("expected io error (cannot connect to coordinator)", err == ERR_GLOBALMODEL_IO);
+
```

Comment:
Q. beyond this, i guess testing the other error codes would be trickier?

Reply:
yeah we need to have a running coordinator instance for this
I can add a task to the "write integration test" story

Reply:
I just saw that there is already a task `Write tests for the C bindings of xaynet_mobile`